src/components/text-input/index.tsx

```diff
@@ -1,29 +1,92 @@
+import { clsx } from 'clsx';
 import Image from 'next/image';
 
+type TextInputIconProps = {
+  iconSrc: string;
+  iconAlt?: string;
+  divider?: boolean;
+  position: 'left' | 'right';
+  onIconClick?: () => void;
+};
+
+export function TextInputIcon(props: TextInputIconProps) {
+  const { iconSrc, iconAlt, divider, onIconClick, position } = props;
+
+  return (
+    <div onClick={onIconClick} className='items-centers relative flex'>
+      <Image
+        src={iconSrc}
+        alt={iconAlt ? iconAlt : ''}
+        width={0}
+        height={0}
+        className={clsx(
+          'mx-[4px] h-auto w-auto',
+          !divider && (position === 'left' ? 'ms-[12px]' : 'me-[12px]'),
+          onIconClick &&
+            'rounded-full p-[4px] hover:cursor-pointer hover:bg-[#ccc] active:bg-[#ccc]',
+        )}
+      />
+      {divider && (
+        <div
+          className={clsx(
+            'absolute h-[23px] w-[1px] bg-[#6969694d]',
+            position === 'left' && 'right-0',
+            position === 'right' && 'left-0',
+          )}
+        ></div>
+      )}
+    </div>
+  );
+}
+
 type TextInputProps = {
   iconSrc?: string;
   iconAlt?: string;
   placeholder: string;
   value: string;
   onInput: (value: string) => void;
   onEnter?: () => void;
+  type?: string;
+  iconPosition?: 'right' | 'left';
+  divider?: boolean;
+  onIconClick?: () => void;
+  fontSize?: number;
+  disabled?: boolean;
+  className?: string;
 };
 
 export default function TextInput(props: TextInputProps) {
-  const { iconSrc, iconAlt, placeholder, value, onInput, onEnter } = props;
+  const {
+    iconSrc,
+    iconAlt,
+    iconPosition = 'left',
+    type,
+    placeholder,
+    value,
+    onInput,
+    onEnter,
+    divider,
+    onIconClick,
+    fontSize = 16,
```

Comment:
```suggestion
    fontSize = 16,
```
This is a component that combine many other component. fontSize is not clear for us to know what is the fontSize mean in this component. You should change props which belong to input -> inputStyle, props belong to container -> containerStyle it will clear